src/Illuminate/Support/Json.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Illuminate\Support;
+
+class Json
+{
+    /**
+     * Compile a PHP expression into a JavaScript object, array or single-quoted string.
+     *
+     * @param  mixed  $expression
+     * @param  int|null  $options
+     * @param  int  $depth
+     * @return string
+     */
+    public static function parse($expression, $options = null, $depth = 512)
+    {
+        if (is_object($expression) || is_array($expression)) {
+            $base64 = base64_encode(json_encode($expression, $options, $depth));
+
+            return "JSON.parse(atob('{$base64}'))";
+        }
+
+        if (is_string($expression)) {
+            return str_replace('"', "'", json_encode($expression));
+        }
```

Comment:
This breaks for any string containing a single quote character.

```php
str_replace('"', "'", json_encode("'"));
// PHP "'''"
// JavaScript '''
//     Invalid syntax since single quotes '' represent a 1 single quote escaped.
//     Another single quote is required to close the expression: ''''
```

Use one of:

1. `json_encode($expression)` without any secondary `str_replace()` calls
   * Blade `{{ }}` -> `e()` HTML encode any single or double quote characters.

     #### Blade template for the component from my earlier comment:

     ```html
     @php
         $foo = '"\'';
     @endphp
     <x-button-js x-data="{ foo: {{ \App\Js::from($foo) }} }">
         Show alert JS
     </x-button-js>
     ```
  
     #### HTML result:
  
     ```html
     <button x-data="{ foo: &quot;\&quot;&#039;&quot; }"
             type="button"
             onclick="data = { foo: &quot;\&quot;&#039;&quot; }; alert(data.foo);">
         Show alert JS
     </button>
     ```
2. `"'".str_replace("'", "\'", str_replace('\\', '\\\\', $expression))."'"` that was suggested earlier

Reply:
I will change to your suggestion.